### PR TITLE
chore(ci): adding YAML-Lint for falco rules 

### LINKED ISSUE #144: Consider adding a yaml-linter for the rules files
**Motivation**

Consider adding a yaml-linter for the rules files, see @Andreagit97 suggestion https://github.com/falcosecurity/rules/pull/142/files#r1308818842. This may help to enforce the new rules style guide consistently.

### PR DESCRIPTION
**Pull request by h4l0gen**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind feature

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

 /area rules

> /area registry

> /area build

> /area documentation

**Proposed rule [maturity level](https://github.com/falcosecurity/rules/blob/main/CONTRIBUTING.md#maturity-levels)**

> Uncomment one (or more) `/area <>` lines (only for PRs that add or modify rules):

> /area maturity-stable

> /area maturity-incubating

> /area maturity-sandbox

> /area maturity-deprecated

**What this PR does / why we need it**:
It checks for syntax in falco rules by use of YAML-Lint 
**Which issue(s) this PR fixes**:

Fixes #144 

**Special notes for your reviewer**:

### CODE REVIEW

**Review by incertum — COMMENTED** [review]
@FedeDP or @maxgio92 could you help with this PR, thanks a bunch in advance!

Thank you @h4l0gen!

**Comment by h4l0gen** [comment]
I have used [YAML-Lint](https://github.com/adrienverge/yamllint) for the YAML linting process. This is the [defaults.yaml](https://github.com/adrienverge/yamllint/blob/master/yamllint/conf/default.yaml) file of YAML-Lint. Please take a look at it and provide suggestions regarding the default values, if any. Thanks!

File: .github/workflows/yaml-lint.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    branches: 
+      - master
+
+jobs: 
+  lintFalcoRules: 
+    name: Yamllint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3 
+      - name: checkout falco-rules
```
Comment (FedeDP) [inline .github/workflows/yaml-lint.yaml:13]:
```suggestion
      - name: Checkout repo
        uses: actions/checkout@v4
```

File: .github/workflows/yaml-lint.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        uses: ibiqlik/action-yamllint@v3
+        with: 
+          files: /rules/*.yaml 
+          yamllint_strict: true
+          yamllint_comment: true
```
Comment (FedeDP) [inline .github/workflows/yaml-lint.yaml:18]: I can't find these 3 input parameters in the yamllint action: https://github.com/ibiqlik/action-yamllint?tab=readme-ov-file#optional-input-parameters

Comment (h4l0gen) [reply .github/workflows/yaml-lint.yaml:18]: yamllint_strict: true
          yamllint_comment: true
 @FedeDP  by default these two options have `false` value. Am not able to figure out in this  https://github.com/ibiqlik/action-yamllint?tab=readme-ov-file#optional-input-parameters what value is set, So just to be safe i set these. If you think we should go with default values then we can remove it.

File: .github/workflows/yaml-lint.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+jobs: 
+  lintFalcoRules: 
+    name: Yamllint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3 
+      - name: checkout falco-rules
+        uses: ibiqlik/action-yamllint@v3
+        with: 
+          files: /rules/*.yaml
```
Comment (h4l0gen) [inline .github/workflows/yaml-lint.yaml:16]: Sorry for that @FedeDP , we should use `file_or_dir: rules/*.yaml` . Let me change it. Should we change it to 
```
steps:
  - uses: actions/checkout@v4
  - name: Checkout repo
    uses: ibiqlik/action-yamllint@v3
    with: 
      file_or_dir: rules/*.yaml
```
WDYT?🤔

File: .github/workflows/yaml-lint.yaml (hunk 2 above)
Comment (FedeDP) [reply .github/workflows/yaml-lint.yaml:18]: I mean, they are not existent input keys, right? We have:
```

    config_file - Path to custom configuration
    config_data - Custom configuration (as YAML source)
    file_or_dir - Enter file/folder (space separated), wildcards accepted. Examples:
        . - run against all yaml files in a directory recursively (default)
        file1.yaml
        file1.yaml file2.yaml
        kustomize/**/*.yaml mychart/*values.yaml
    format - Format for parsing output [parsable,standard,colored,github,auto] (default: parsable)
    strict - Return non-zero exit code on warnings as well as errors [true,false] (default: false)
    no_warnings - Output only error level problems [true,false] (default: false)

```

Comment (h4l0gen) [reply .github/workflows/yaml-lint.yaml:18]: Yes @FedeDP they aren't existing input keys. I think we should remove these two points ?
```
      yamllint_strict: true
      yamllint_comment: true
```

**Comment by h4l0gen** [comment]
Hey @FedeDP  I have done the recommended changes. Please take a look. Thanks!

File: .github/workflows/yaml-lint.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    branches: 
+      - master
+
+jobs: 
+  lintFalcoRules: 
+    name: Yamllint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+      - uses: actions/checkout@v4
```
Comment (FedeDP) [inline .github/workflows/yaml-lint.yaml:13]:
```suggestion
      - name: Checkout repo
        uses: actions/checkout@v4
```

File: .github/workflows/yaml-lint.yaml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+name: Yamllint Github Actions
+on: 
+  pull_request:
+    branches: 
+      - master
```
Comment (FedeDP) [inline .github/workflows/yaml-lint.yaml:5]:
```suggestion
      - main
```

**Comment by FedeDP** [comment]
>  Yamllint Github Actions / Yamllint (pull_request) Failing after 6s 

We've got lots of issues related to line length and trailing spaces :)

**Comment by h4l0gen** [comment]
@FedeDP Yes that's alot😅, I guess we should create an issue for removing trailing-spaces in falco-rules.yaml file and for line-length, limit is only 80 chars and length in falco-rules goes upto 377 chars, so either we change its limit or we can write in [multiple strings ](https://stackoverflow.com/questions/3790454/how-do-i-break-a-string-in-yaml-over-multiple-lines/21699210#21699210) . IMO trailing space is not a big issue so we can disable the rules too. WDYT about it 🤔.

**Comment by FedeDP** [comment]
I agree, we should tackle them in a subsequent PR; trailing spaces should be the simplest to fix, so we can tackle it in a follow up PR; instead for the line length we can think of splitting them on multiple lines, or disable the rule!
cc @leogr @incertum  wdyt?

**Comment by h4l0gen** [comment]
Hii @leogr @FedeDP @incertum If you all agree then i would like to create a PR for removing trailing spaces and handling issue of line length with multiple lines.

**Comment by h4l0gen** [comment]
@FedeDP i rebased this PR.

**Review by FedeDP — APPROVED** [review]
/approve

**Comment by FedeDP** [comment]
Let's merge this one so that we have a yaml-lint CI; then @h4l0gen will open a new PR with the fixes to make the new CI happy :) 
Thanks!

**Pull request merged**